### ISSUE TITLE
Explorer reporting balance 0 but transactions add up to much more than 0

### ISSUE THREAD

**Issue by LeeAlephium** [alephium/explorer-backend#319]
Check out this address on testnet: https://testnet.alephium.org/addresses/1HRr1oBBSPJqeodrSnpKGiWv9iSpWiND1zZNPdL4tL8Ev

It received 10 ALPH but only sent out 0.002 ALPH. How come the balance is now zero?

I can explain why it's zero, but not why there are missing transactions: I had sent additional transactions to drain this address. So for some reason explorer is just not showing the rest of the transactions.

/cc @tdroxler

**Comment by tdroxler** [comment]
I'll have a look, thx for reporting

**Comment by tdroxler** [comment]
damn, I just synced locally and everything is fine :/ I'll try to access our aws testnet database

**Comment by tdroxler** [comment]
I got access to the DB, there are indeed a lot of forks, you have somethink like 500 outputs that are not in the main chain, but it's not the issue.

there is a bug in [the balance query of v1.6.1](https://github.com/alephium/explorer-backend/blob/v1.6.1/app/src/main/scala/org/alephium/explorer/persistence/queries/TransactionQueries.scala#L371-L378) we miss `inputs.main_chain=true`, if we add this we get the correct result:
```
alphexplorer=> SELECT outputs.amount, outputs.lock_time
alphexplorer-> FROM outputs
alphexplorer-> LEFT JOIN inputs ON inputs.main_chain = true AND outputs.key = inputs.output_ref_key
alphexplorer-> WHERE outputs.main_chain = true AND outputs.address ='1HRr1oBBSPJqeodrSnpKGiWv9iSpWiND1zZNPdL4tL8Ev' AND inputs.block_hash IS NULL;
       amount        | lock_time
---------------------+-----------
 1000000000000000000 |
 8998000000000000000 |
(2 rows)
```
[This is something fixed in v1.6.9](https://github.com/alephium/explorer-backend/blob/v1.6.9/app/src/main/scala/org/alephium/explorer/persistence/queries/TransactionQueries.scala#L418-L435), the version that run on the mainnet.
We will update to that version and I'll close the issue once we see the balance is correct.

**Comment by tdroxler** [comment]
v1.6.9 is deployed and look much better
https://testnet.alephium.org/addresses/1HRr1oBBSPJqeodrSnpKGiWv9iSpWiND1zZNPdL4tL8Ev
